AirCasting/ABConnector/BluetoothManager.swift

```diff
@@ -110,15 +110,19 @@ extension BluetoothManager: CBCentralManagerDelegate {
     func centralManager(_ central: CBCentralManager, didConnect peripheral: CBPeripheral) {
         // Here's code for getting data from AB.
         peripheral.delegate = self
-        peripheral.discoverServices(nil)
         if mobilePeripheralSessionManager.activeSessionInProgressWith(peripheral) {
-            DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(4)) {
+            var characteristicsHandle: Any?
+            characteristicsHandle = NotificationCenter.default.addObserver(forName: .discoveredCharacteristic, object: nil, queue: .main) { notification in
+                guard notification.userInfo?["peripheral uuid"] as! UUID == peripheral.identifier else { return }
```

Comment:
we have a special place to store keys instead of typing a string (the same file as `.discoveredCharacteristic`, the `AirCastingNotificationKeys` enum). This prevents us from making a typo and wasting 3 days to find why stuff is not working ;)